### PR TITLE
apigw clean up the invalid json parser

### PR DESCRIPTION
**Pull request by cloutierMat**
## Motivation

In an attempt to move fast to fix the issue, with @bentsku. we created #11967.

The naming was done hastily and we will attempt to clean it up a little in this pr.

## Changes

- clean up naming
- simplify logic
- adding a few tests

### CODE REVIEW

File: localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py [hunk 1]
```diff
@@ -66,45 +66,37 @@ def parse_invalid_json(self, body: str) -> dict:
...
+                key = key or '""'
+                value = value or '""'
+
+                assert (leading_key_char := key[0]) not in "[{"
+                if leading_key_char in "'\"":
+                    assert len(key) >= 2
+                    assert key[-1] == leading_key_char
+
+                if key == "statusCode":
+                    statuscode = int(value)
```
Comment (bentsku) [inline localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py:84]: I think we had the mistake in our current implementation too, but here we directly check against the `stateCode` value without regards to single or double quotes it seems (except of the leftover second `if` check)

I think we need to still address that 🤔

File: tests/unit/services/apigateway/test_mock_integration.py [hunk 2]
```diff
@@ -61,6 +61,9 @@ def test_custom_parser(self, create_default_context):
             "{statusCode: 200, foo}: [ba r]}",
             "{statusCode: 200, foo'}: [ba r]}",
             "{statusCode: 200, }foo: [ba r]}",
+            "{statusCode: 200, }foo: ''}",
+            '{statusCode: 200, " ": " "}',
+            '{statusCode: 200, "": ""}',
```
Comment (bentsku) [inline tests/unit/services/apigateway/test_mock_integration.py:66]: we could also add a few cases with `statusCode` in single and double quotes, and maybe a few invalid with non closing quotes or something similar?

**Review by bentsku — COMMENTED** [review]
Thanks a lot for cleaning up our... best? code! 😄 I think we still can improve the quote handling of the key, especially to match the `statusCode` value, but then we should be good to go! Thanks a lot again 🙏

File: localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py (hunk 1 above)
Comment (cloutierMat) [reply localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py:84]: oh yes! I must have been tired when I wrote made the cleanup! The goal of moving it under the quotes assertions, was to solve this very issue! But it isn't removing the quotes anymore as we originally had :facepalm:

File: tests/unit/services/apigateway/test_mock_integration.py (hunk 2 above)
Comment (cloutierMat) [reply tests/unit/services/apigateway/test_mock_integration.py:66]: Good thinking! that is indeed the most important objective :rofl:

File: localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py [hunk 3]
```diff
@@ -63,48 +63,46 @@ def parse_invalid_json(self, body: str) -> dict:
...
-                assert v
-
-                if k == "statusCode":
-                    statuscode = int(v)
+            key_values = [s.split(":", maxsplit=1) for s in pairs if s]
+            for key_value in key_values:
+                assert len(key_value) == 2
+                key, value = [el.strip() for el in key_value]
+                key = convert_null_value(key)
+                value = convert_null_value(value)
```
Comment (bentsku) [inline localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py:82]: nit: could probably be simplified with:
```suggestion
                key, value = [convert_null_value(el.strip()) for el in key_value]
```

I think we could even put the `strip()` call inside `convert_null_value`.

**Review by bentsku — APPROVED** [review]
LGTM! Just a minor simplification, but this is already much cleaner than before 🙏 Thanks a lot for addressing the comments and adding test cases! Really, this is such a weird case...

File: localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py (hunk 3 above)
Comment (bentsku) [reply localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/mock.py:82]: Oops, missed it but we could probably add type hints for `convert_null_value(value: str) -> str:`

**Pull request merged**